### DISCUSSION TITLE
How to integrate the jar with the OTEL agent JAR?

### DISCUSSION THREAD

**Discussion by tarnisheddd** [open-telemetry/opentelemetry-java-contrib#566] [Q&A]
Hi Team, I wanted to utilize the JFR events and streaming with OTEL Agent, but I am not able to integrate them. Can someone help on how to do this?

**Comment by mateuszrzeszutek** [comment]
Hey @vaibhavcymk ,

`jfr-streaming` should be pretty easy to use:
```
JfrMetrics.enable(GlobalOpenTelemetry.getMeterProvider());
```
For `jfr-events` you'll need a custom [javaagent extension](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension) that'll install the `SpanProcessor` and the `ContextStorage` wrapper.
